sig-and-wg/sig-core/decisions/dr-020-Feature_gates.md

```diff
@@ -0,0 +1,43 @@
+# Decision 020: Feature Gates in Kyma
+
+Created on 2018-09-28 by Ahmed Abdalla(@Abd4llA).
+
+## Decision log
+
+| Name| Description |
+|-----------------------|------------------------------------------------------------------------------------|
+| Title | Introducing feature gates in Kyma |
+| Ultimate decision maker(s) | Kyma Council |
+| Due date | 2018-10-10 |
+| Input provider(s) | POs, Architects |
+| Group(s) affected by the decision | Kyma developers, Kyma users |
+| Decision type | Binary |
+| Earliest date to revisit the decision | 2019-01-10 |
+
+## Context
+
+Kyma release strategy and branching model defines master should always be in a
+"releasable" state and Kyma follows a cactus branching model with releases
+branches. So in principal Kyma is following a trunk based development model with
+no long living feature branches. A core technique of applying trunk based
+development is using feature toggles, and in Kyma case introducing Kyma wide
+feature gates is needed and not just per component.
+
+This way when introducing major changes such as rebasing Kyma on Knative we can
+hide all modifications behind a kyma wide feature flag.
```

Comment:
For now this proposal only shows one use-case: installation on knative.
Proposed solution assumes a non-trivial implementation effort: building an API, golang library, etc.
If this is only to support one use case then, for me, it's not justifiable effort and we're good with current solution: helm global value to control conditional behaviour, switching components on the list.
I would even say, that we can go with current approach until the number of "features" does not exceed some limit - my gut feeling is ten (10).
Below this number we can live with current solution without much problems.
If, and only if, this becomes very hard to maintain and document (over 10) we can consider building such an Feature Gate solution.